aten/src/ATen/native/sparse/ValidateCompressedIndicesCommon.h

```diff
@@ -241,52 +266,79 @@ void _validate_compressed_sparse_indices_kernel(
 
     const auto batch_dims = cidx.sizes().slice(0, cidx.dim() - 1);
     const auto batch_count = indexCount(batch_dims);
-    const auto batch_idx = at::arange(batch_count, cidx.options()).view(batch_dims).unsqueeze_(-1);
+    const auto batch_idx =
+        at::arange(batch_count, cidx.options()).view(batch_dims).unsqueeze_(-1);
+
+    const Tensor batch_shape = at::tensor(batch_dims, idx.options());
+    const Tensor batch_idx_stride =
+        at::tensor(idx.strides().slice(0, idx.dim() - 1), idx.options());
+    const Tensor all_idx_offsets = at::arange(
+        batch_idx_stride.mul(batch_shape.sub(1)).sum().add(1).item(),
+        idx.options());
+    const Tensor idx_offsets =
+        all_idx_offsets.reshape({-1, 1})
+            .div(batch_idx_stride, "floor") // TODO: support zero strides
+            .remainder(batch_shape)
```

Comment:
Is it easy to fix? Do we get an error message and not a segfault if there are zero strides?

Reply:
`make_offset_calculator` appears to be CUDA specific but there seems to be analogous code for the CPU. I'll investigate that..

Re zero strides: I think we should support it because otherwise, test_bmm needs to use an unorthodox way of constructing CSR tensors. I suggest leaving it as a follow-up PR within this ghstack as in addition to generalizing the above, the generate_simple_inputs need to be updated accordingly and it is unpredictable how other existing tests behave on zero-strided inputs.

Reply:
Then, I guess, it would be great to just error out, and then remove it in a follow-up as to avoid any potential segfaults...

Reply:
Fixed, the latest commit supports zero strides.